src/Microsoft.VisualBasic/tests/VB/FileIOTests.vb

```diff
@@ -0,0 +1,897 @@
+        End Sub
+
+        <Fact>
+        Public Shared Sub GetDirectories_Directory_SearchOption()
+            Using TestBase As New FileIOTests
+                Dim DirectoryList As ReadOnlyCollection(Of String) = FileSystem.GetDirectories(TestBase.TestDirectory, SearchOption.SearchTopLevelOnly)
+                Assert.True(DirectoryList.Count = 0)
+                For i As Integer = 0 To 5
+                    IO.Directory.CreateDirectory(IO.Path.Combine(TestBase.TestDirectory, $"GetDirectories_Directory_SearchOptionNewSubDirectory{i}"))
+                Next
+                DirectoryList = FileSystem.GetDirectories(TestBase.TestDirectory, SearchOption.SearchTopLevelOnly)
+                Assert.True(DirectoryList.Count = 6)
+                For i As Integer = 0 To 5
+                    Assert.True(DirectoryList.Contains(IO.Path.Combine(TestBase.TestDirectory, $"GetDirectories_Directory_SearchOptionNewSubDirectory{i}")))
+                Next
+                IO.Directory.CreateDirectory(IO.Path.Combine(TestBase.TestDirectory, $"GetDirectories_Directory_SearchOptionNewSubDirectory0", $"NewSubSubDirectory"))
+                DirectoryList = FileSystem.GetDirectories(TestBase.TestDirectory, SearchOption.SearchTopLevelOnly)
+                Assert.True(DirectoryList.Count = 6)
+                DirectoryList = FileSystem.GetDirectories(TestBase.TestDirectory, SearchOption.SearchAllSubDirectories)
+                Assert.True(DirectoryList.Count = 7)
+            End Using
+        End Sub
+
+        <Fact>
+        Public Shared Sub GetDirectories_Directory_SearchOption_Wildcards()
+            Using TestBase As New FileIOTests
+                Dim DirectoryList As ReadOnlyCollection(Of String) = FileSystem.GetDirectories(TestBase.TestDirectory, SearchOption.SearchTopLevelOnly, "*")
+                Assert.True(DirectoryList.Count = 0)
+                Dim CreatedDirectories As New List(Of String)
+                For i As Integer = 0 To 5
+                    CreatedDirectories.Add(IO.Directory.CreateDirectory(IO.Path.Combine(TestBase.TestDirectory, $"NewSubDirectory00{i}")).Name)
+                Next
+                DirectoryList = FileSystem.GetDirectories(TestBase.TestDirectory, SearchOption.SearchTopLevelOnly, "*000", "*001")
+                Assert.True(DirectoryList.Count = 2, $"Search results Expected 2 Actual {DirectoryList.Count} DirectoryList = {DirectoryListToString(DirectoryList)} Created List = {DirectoryListToString(New ReadOnlyCollection(Of String)(CreatedDirectories))}")
+                For i As Integer = 0 To 1
+                    Dim DirectoryName As String = IO.Path.Combine(TestBase.TestDirectory, $"NewSubDirectory00{i}")
+                    Assert.True(DirectoryList.Contains(DirectoryName), $"{DirectoryName} Is missing from Wildcard Search")
+                Next
+                IO.Directory.CreateDirectory(IO.Path.Combine(TestBase.TestDirectory, $"NewSubDirectory000", $"NewSubSubDirectory000"))
```

Comment:
Consider adding under `000` under `001` instead. Or add `000` and `001` under the same sub directory.

Reply:
Sorry I am not following this issue or what needs to change